cpp/distributed/dist-mnist.cpp

```diff
@@ -120,14 +115,7 @@ int main(int argc, char* argv[]) {
       // overlaps synchronizing parameters and computing gradients in backward
       // pass
       for (auto& param : model->named_parameters()) {
-        MPI_Allreduce(
-            MPI_IN_PLACE,
-            param.value().grad().data_ptr(),
-            param.value().grad().numel(),
-            mpiDatatype.at(param.value().grad().scalar_type()),
-            MPI_SUM,
-            MPI_COMM_WORLD);
-
+        pg.allreduce(param.value().grad());
```

Comment:
How to add the AllreduceOptions for the SUM operator?

Reply:
`AllreduceOptions` is a second optional argument for the `allreduce` function.
You can create the AllreduceOptions struct and set the `reduceOp = ReduceOp::SUM`, but this is already the default so you should be able to leave this as is.

Reply:
Alright. Also adding the waiting mechanism as stated in [https://github.com/pytorch/pytorch/blob/master/torch/lib/c10d/test/ProcessGroupMPITest.cpp](https://github.com/pytorch/pytorch/blob/master/torch/lib/c10d/test/ProcessGroupMPITest.cpp)